utilities/infra.py

```diff
@@ -1536,3 +1537,17 @@ def verify_image_info(image_url, generated_pulled_secret, nodes_cpu_architecture
             f"--registry-config={generated_pulled_secret} --filter-by-os={nodes_cpu_architecture}"
         ),
     )
+
+
+def assert_os_version_mismatch_in_vm(vm: utilities.virt.VirtualMachineForTests, expected_os: str) -> None:
+    vm_os = vm.ssh_exec.os.release_str.lower()
+    match = re.match(r"(?P<os_name>[a-z]+)(?:\.stream|\.)?(?P<os_ver>[0-9]+)?", expected_os)
+    if match:
+        expected_os_params = match.groupdict()
```

Comment:
In this scenario, since we are looking for os name and version only,  why not use simply group and direct assignments